test/index.js

```diff
@@ -3197,6 +3197,40 @@ describe('list', function() {
 
   });
 
+  describe('reduce', function() {
+
+    it('is a ternary function', function() {
+      eq(typeof S.reduce, 'function');
+      eq(S.reduce.length, 3);
+    });
+
+    it('type checks its arguments', function() {
+      assert.throws(function() { S.reduce('xxx'); },
+                    errorEq(TypeError,
+                            '‘reduce’ expected a value of type Function ' +
+                            'as its first argument; received "xxx"'));
+    });
+
+    it('folds over lists with the supplied accumulator', function() {
+      eq(S.reduce(S.add, 0, [1, 2, 3, 4, 5]), 15);
+      eq(S.reduce(S.add, 0, []), 0);
+      eq(S.reduce(S.lift2(S.add), S.Just(0),
+                  [S.Just(1), S.Just(2), S.Just(3), S.Just(4), S.Just(5)]),
+         S.Just(15));
+    });
+
+    it('dispatches to a reduce method if present', function() {
+      eq(S.reduce(function(a, b) { return a + b; }, 10, S.Just(5)), 15);
```

Comment:
Is there a reason to use a lambda rather than `S.add` here? Are you asserting that the function needn't be curried?